stable/redis-ha/templates/redis-haproxy-service.yaml

```diff
@@ -0,0 +1,31 @@
+{{- if .Values.haproxy.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "redis-ha.fullname" . }}-haproxy
+  labels:
+{{ include "labels.standard" . | indent 4 }}
+  annotations:
+  {{- if .Values.haproxy.service.annotations }}
+{{ toYaml .Values.haproxy.service.annotations | indent 4 }}
+  {{- end }}
+spec:
+  type: {{ default "ClusterIP" .Values.haproxy.service.type }}
+  {{- if or (not .Values.haproxy.service.type) (eq .Values.haproxy.service.type "ClusterIP") }}
+  clusterIP: None
+  {{- end }}
+  ports:
+  - name: haproxy
+    port: {{ .Values.redis.port }}
+    protocol: TCP
+    targetPort: redis
+{{- if .Values.exporter.enabled }}
+  - name: exporter-port
+    port: {{ .Values.haproxy.exporter.port }}
+    protocol: TCP
+    targetPort: exporter-port
```

Comment:
Ideally this would have the sentinel port as well so HAproxy could balance requests across sentinels as well (the redis-ha headless service won't balance and simply send request to the first sentinel endpoint).

Reply:
Why would we need to balance sentinel ports? Ideally, this solution doesn't need Sentinel at all. It circumvents the need for that altogether.

If you think people might use sentinel client beyond this though, I can add it.

Reply:
I don't think it's essential to the primary goal of this PR but I do think the issue would eventually come up (one service to rule them all instead of needing two). It can be left out of this PR and always added at a later date.